test/e2e/validation.go

```diff
@@ -43,159 +46,112 @@ const (
-		if err != nil {
-			debug("Error getting TopoLVM node daemonset %s: %s\n", topolvmNodeDaemonSetName, err.Error())
-		}
-		return err == nil
-	}, timeout, interval).Should(BeTrue())
-	debug("TopoLVM node daemonset found\n")
-
-	// checking for the ready status
-	Eventually(func() bool {
-		err := crClient.Get(ctx, types.NamespacedName{Name: topolvmNodeDaemonSetName, Namespace: installNamespace}, &ds)
-		if err != nil {
-			debug("Error getting TopoLVM node daemonset %s: %s\n", topolvmNodeDaemonSetName, err.Error())
-		}
-		return ds.Status.DesiredNumberScheduled == ds.Status.NumberReady
-	}, timeout, interval).Should(BeTrue())
-	debug("TopoLVM node pods are ready\n")
-
-	return nil
+func validateTopolvmNode(ctx context.Context) bool {
+	By("validating the TopoLVM Node DaemonSet")
+	return validateDaemonSet(ctx, types.NamespacedName{Name: topolvmNodeDaemonSetName, Namespace: installNamespace})
 }
 
 // function to validate vg manager resource.
-func validateVGManager(ctx context.Context) error {
-	ds := appsv1.DaemonSet{}
-
-	Eventually(func() bool {
-		err := crClient.Get(ctx, types.NamespacedName{Name: vgManagerDaemonsetName, Namespace: installNamespace}, &ds)
-		if err != nil {
-			debug("Error getting vg manager daemonset %s: %s\n", vgManagerDaemonsetName, err.Error())
-		}
-		return err == nil
-	}, timeout, interval).Should(BeTrue())
-	debug("VG manager daemonset found\n")
-
-	return nil
+func validateVGManager(ctx context.Context) bool {
+	By("validating the vg-manager DaemonSet")
+	return validateDaemonSet(ctx, types.NamespacedName{Name: vgManagerDaemonsetName, Namespace: installNamespace})
 }
 
 // function to validate TopoLVM Deployment.
-func validateTopolvmController(ctx context.Context) error {
-	dep := appsv1.Deployment{}
+func validateTopolvmController(ctx context.Context) bool {
+	By("validating the TopoLVM controller deployment")
+	return validateDeployment(ctx, types.NamespacedName{Name: topolvmCtrlDeploymentName, Namespace: installNamespace})
+}
 
-	Eventually(func() bool {
-		err := crClient.Get(ctx, types.NamespacedName{Name: topolvmCtrlDeploymentName, Namespace: installNamespace}, &dep)
-		if err != nil {
-			debug("Error getting TopoLVM controller deployment %s: %s\n", topolvmCtrlDeploymentName, err.Error())
+func validateDaemonSet(ctx context.Context, name types.NamespacedName) bool {
+	return Eventually(func(ctx context.Context) error {
+		ds := &appsv1.DaemonSet{}
+		if err := crClient.Get(ctx, name, ds); err != nil {
+			return err
 		}
-		return err == nil
-	}, timeout, interval).Should(BeTrue())
+		isReady := ds.Status.DesiredNumberScheduled == ds.Status.NumberReady
+		if !isReady {
+			return fmt.Errorf("the DaemonSet %s is not considered ready", name)
+		}
+		return nil
+	}, timeout, interval).WithContext(ctx).Should(Succeed())
+}
 
-	debug("TopoLVM controller deployment found\n")
-	return nil
+func validateDeployment(ctx context.Context, name types.NamespacedName) bool {
```

Comment:
I know it's returning the result of the eventually but I'd still rather see this in the above functions and reduce the amount of hopping around we need to do in the code. In typical code I'd be more okay with it if it made it easier to unit test but that isn't the case here since we are in e2e

Reply:
in general I agree hopping is bad, but currently there is a clear layer now where something fails because I passed through the ginkgo context. It will give you an exact stack trace showing you where things failed with each call pass being shown as well. If you still think we should move it out I will do so of course!